integrations/jira/jira/client.py

```diff
@@ -43,25 +54,95 @@ def __init__(self, jira_url: str, jira_email: str, jira_token: str) -> None:
 
     @staticmethod
     def _generate_base_req_params(
-        maxResults: int = 0, startAt: int = 0
+        maxResults: int = 50, startAt: int = 0
     ) -> dict[str, Any]:
         return {
             "maxResults": maxResults,
             "startAt": startAt,
         }
 
-    async def _get_paginated_projects(self, params: dict[str, Any]) -> dict[str, Any]:
+    async def _make_paginated_request(
+        self, url: str, params: dict[str, Any] = {}
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        params = {**params, **self._generate_base_req_params()}
+        is_last = False
+        while not is_last:
+            response = await self.client.get(url, params=params)
+            response.raise_for_status()
+            response_data = response.json()
+            values = response_data["values"]
+            yield values
+            is_last = response_data["isLast"]
+            start = response_data["startAt"] + response_data["maxResults"]
+            params = {**params, "startAt": start}
+        return
+
+    async def _get_single_item_request(
+        self, url: str, params: dict[str, Any] = {}
+    ) -> dict[str, Any]:
+        response = await self.client.get(url, params=params)
+        response.raise_for_status()
+        return response.json()
```

Comment:
Please add logs to this function

Reply:
I also don't see where you're using this function. Please use it in the webhook event handler